torch/csrc/jit/script/init.cpp

```diff
@@ -121,17 +156,43 @@ struct VISIBILITY_HIDDEN PythonValue : public SugaredValue {
 
     // Release the function object so we can wrap it in a PythonOp
     py::object func = self;
-    std::string cconv(inputs.size(), 'd');
-    Node* new_node = m.graph()->insertNode(m.graph()->createPythonOp(
-      THPObjectPtr(func.release().ptr()), cconv, {}));
+
```

Comment:
Take a look at how torch.jit.script functions are handled in `toSugaredValue` (insert some debugging code if you want to trace how it works). In particular, they are treated as ModuleValues. Weak script should be treated the same way. In `toSugaredValue` if you find a weak script function, convert it to a ScriptModule (and cache the script module). There does not need to be any code changes for the calling a function path.